### ISSUE TITLE
Compare TA Indicators with ML Indicators

### ISSUE THREAD

**Issue by slavakurilyak** [produvia/kryptos#77]
# Goal

As a developer, I want to compare Technical Analysis (TA) indicators against Machine Learning (ML) indicators, so that I can see the best performing strategies on Kryptos.

# Inspiration

## Machine Learning Model

Running `$ strat -ml xgboost` on `config.json`

```json
# config.json
{
   "EXCHANGE": "bitfinex",
   "ASSET": "btc_usd",
   "DATA_FREQ": "daily",
   "HISTORY_FREQ": "1d",
   "CAPITAL_BASE": 5000,
   "BASE_CURRENCY": "usd",
   "START": "2015-04-01",
   "END": "2018-06-07"
}
```

produced the following graph:

<img width="1552" alt="screen shot 2018-06-08 at 10 11 02 pm" src="https://user-images.githubusercontent.com/6625584/41187914-77f58774-6b69-11e8-8871-9f220cc62b32.png">

Here is the `backtest_summary.csv`

```csv
# backtest_summary.csv
,Backtest
start_date,2015-04-01
end_date,2018-06-07
backtest_minutes,0.0
backtest_days,1163.0
backtest_weeks,166.14285714285714
number_of_trades,79
average_trades_per_week_avg,0.4754944110060189
average_trade_amount_usd,25.582588227407154
initial_capital,5000.0
ending_capital,12444.44489328955
net_profit,7444.44489328955
net_profit_pct,148.888897865791
average_daily_profit,6.401070415554213
average_daily_profit_pct,0.12802140831108427
average_exposure,858.5830145073105
average_exposure_pct,10.34908635604034
net_risk_adjusted_return_pct,8.670617479617242
max_drawdown_pct_catalyst,-43.38881070084797
max_daily_drawdown_pct,-11.179883683118964
max_weekly_drawdown_pct,-17.272549369097096
sharpe_ratio_avg,0.6942761152089412
std_rolling_10_day_pct_avg,0.009477632765067757
std_rolling_100_day_pct_avg,0.037640182736585044
number_of_simulations,1164
```

The `xgboost_confussion_matrix.txt` showed:

```txt
Accuracy: 0.4836769759450172
Coefficient Kappa: 0.174298542554849
Classification Report:
             precision    recall  f1-score   support

       KEEP       0.62      0.60      0.61       568
         UP       0.39      0.50      0.44       344
       DOWN       0.29      0.20      0.24       252

avg / total       0.48      0.48      0.48      1164

Confussion Matrix:
[[340 158  70]
 [120 173  51]
 [ 89 113  50]]
```

## Technical Analysis Model

Running `$ strat -ta sma` on the same `config.json` (see ^ above) produced the following graph:

<img width="1440" alt="screen shot 2018-06-08 at 10 23 05 pm" src="https://user-images.githubusercontent.com/6625584/41188029-03edbefc-6b6c-11e8-8923-4eb24a9bac55.png">

Here is the `backtest_summary.csv`

```csv
# backtest_summary.csv
,Backtest
start_date,2015-04-01
end_date,2018-06-07
backtest_minutes,0.0
backtest_days,1163.0
backtest_weeks,166.14285714285714
number_of_trades,16
average_trades_per_week_avg,0.09630266552020636
average_trade_amount_usd,69.37820330563477
initial_capital,5000.0
ending_capital,8218.92955794566
net_profit,3218.9295579456593
net_profit_pct,64.3785911589132
average_daily_profit,2.7677812192138087
average_daily_profit_pct,0.05535562438427618
average_exposure,916.8338046602403
average_exposure_pct,11.765012716730135
net_risk_adjusted_return_pct,3.5109193635574205
max_drawdown_pct_catalyst,-41.694296557111585
max_daily_drawdown_pct,-10.468402736447736
max_weekly_drawdown_pct,-20.361011437940927
sharpe_ratio_avg,0.7790858915079671
std_rolling_10_day_pct_avg,0.007685927117612323
std_rolling_100_day_pct_avg,0.03348039989489331
number_of_simulations,1164
```

In other words, machine learning model outperformed technical analysis.
